src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -37,33 +37,43 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
     -->
 
-  <PropertyGroup Condition="'$(NoBuild)' != 'true'">
-    <_BeforePublishTargets>Build</_BeforePublishTargets>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(NoBuild)' == 'true'">
+  <PropertyGroup>
     <!-- We still need to resolve references even if we are not building during publish. -->
     <!-- BuildOnlySettings are required for RAR to find satellites and dependencies -->
-    <_BeforePublishTargets>
+    <_BeforePublishNoBuildTargets>
       BuildOnlySettings;
       _PreventProjectReferencesFromBuilding;
       ResolveReferences;
       PrepareResourceNames;
       ComputeIntermediateSatelliteAssemblies;
-    </_BeforePublishTargets>
+    </_BeforePublishNoBuildTargets>
+
+    <_CorePublishTargets>
+      PrepareForPublish;
+      ComputeAndCopyFilesToPublishDirectory;
+      GeneratePublishDependencyFile;
+      GeneratePublishRuntimeConfigurationFile;
+    </_CorePublishTargets>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(NoBuild)' != 'true'">
+    <_BeforePublishTargets>Build</_BeforePublishTargets>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(NoBuild)' == 'true'">
+    <_BeforePublishTargets>$(_BeforePublishNoBuildTargets)</_BeforePublishTargets>
   </PropertyGroup>
 
   <Target Name="Publish"
-          Condition="$(IsPublishable) == 'true'"
-          DependsOnTargets="$(_BeforePublishTargets);
-                            PrepareForPublish;
-                            ComputeAndCopyFilesToPublishDirectory;
-                            GeneratePublishDependencyFile;
-                            GeneratePublishRuntimeConfigurationFile">
-
-    <!-- Ensure there is minimal verbosity output pointing to the publish directory and not just the
-         build step's minimal output. Otherwise there is no indication at minimal verbosity of where
-         the published assets were copied. -->
+          Condition="$(IsPublishable) == 'true' and '$(NoBuild)' != 'true'"
```

Comment:
@nguerrera Only add an extra condition is added based on the "WIP" and let PackTool depends on both of them. But only one of them will be run according to the flag at that time

Reply:
I don't really understand what you wrote here. We can't condition Publish on NoBuild because we need /t:Publish /p:NoBuild=true to work. I think you're acknowledging that and that this isn't final, but I'm not entirely sure. ;)

Reply:
Ah... you are right. I still need a target to "re direct"